File: examples/InputOutput.hs
-- --------------
-- InputOutput.hs
-- --------------

{-
(>>)     :: IO a   -> IO b        -> IO b      -- 'then' operator
(>>=)    :: IO a   -> (a -> IO b) -> IO b      -- 'bind' operator
getLine  ::                          IO String
putStrLn :: String                -> IO ()
-}

main :: IO ()
main =
    putStrLn "Enter: " >>
    getLine            >>=
    putStrLn

{-
Enter:
abc
abc
-}

File: examples/InputOutput.py
#!/usr/bin/env python

# --------------
# InputOutput.py
# --------------

import sys
import types

print "Enter: "

assert type(sys)       is types.ModuleType
assert type(sys.stdin) is file

s = sys.stdin.readline()
assert type(s) is str

print s, # trailing "," avoids a newline

"""
Enter:
abc
abc
"""
